File: frontend/src/utils/vehicleUtils.ts
export const VEHICLE_TYPES = ['TRUCK', 'VAN', 'CONTAINER'] as const;

export const VEHICLE_STATUSES = ['AVAILABLE', 'IN_USE', 'MAINTENANCE', 'ARCHIVED'] as const;

export const VEHICLE_FILTER_SORT = [
  'NEWEST',
  'OLDEST',
  'CAPACITY_HIGH',
  'CAPACITY_LOW',
] as const;

export const translateVehicleType = (value: string): string => {
  switch (value) {
    case 'TRUCK': return 'Xe tải';
    case 'VAN': return 'Xe van';
    case 'CONTAINER': return 'Xe container';
    default: return value;
  }
};

export const translateVehicleStatus = (value: string): string => {
  switch (value) {
    case 'AVAILABLE': return 'Sẵn sàng';
    case 'IN_USE': return 'Đang sử dụng';
    case 'MAINTENANCE': return 'Đang bảo trì';
    case 'ARCHIVED': return 'Ngừng hoạt động';
    default: return value;
  }
};

export const translateVehicleFilterSort = (value: string): string => {
  switch (value) {
    case 'NEWEST': return 'Mới nhất';
    case 'OLDEST': return 'Cũ nhất';
    case 'CAPACITY_HIGH': return 'Tải trọng cao nhất';
    case 'CAPACITY_LOW': return 'Tải trọng thấp nhất';
    default: return value;
  }
};

File: backend/src/main/java/com/logistics/utils/VehicleUtils.java
package com.logistics.utils;

import com.logistics.enums.VehicleStatus;
import com.logistics.enums.VehicleType;

public class VehicleUtils {

    public static String translateVehicleType(VehicleType value) {
        if (value == null) return "";

        return switch (value) {
            case TRUCK -> "Xe tải";
            case VAN -> "Xe van";
            case CONTAINER -> "Xe container";
            default -> value.name();
        };
    }

    public static String translateVehicleStatus(VehicleStatus value) {
        if (value == null) return "";

        return switch (value) {
            case AVAILABLE -> "Sẵn sàng";
            case IN_USE -> "Đang sử dụng";
            case MAINTENANCE -> "Đang bảo trì";
            case ARCHIVED -> "Ngừng hoạt động";
            default -> value.name();
        };
    }
}